pkg/cgroups/cgroupstats.go

```diff
@@ -368,72 +379,73 @@ func GetNumaStats(cgroupPath string) (NumaStat, error) {
 	// hierarchical_anon=46096 N0=12597 N1=18890 N2=283 N3=14326
 	// hierarchical_unevictable=20 N0=0 N1=0 N2=0 N3=20
 
-	entry := path.Join(cgroupPath, cgroupEntry)
-	lines, err := readCgroupFileLines(entry)
-	if err != nil {
-		return NumaStat{}, err
-	}
+	var nline *NumaLine
 
 	result := NumaStat{}
-	for _, line := range lines {
-		split := strings.Split(line, " ")
-		if len(line) < 2 {
-			return NumaStat{}, fmt.Errorf("error parsing file %s", entry)
-		}
-
-		keytotal := strings.Split(split[0], "=")
-		if len(keytotal) != 2 {
-			return NumaStat{}, fmt.Errorf("error parsing file %s", entry)
-		}
-		key, tot := keytotal[0], keytotal[1]
-
-		total, err := strconv.ParseInt(tot, 10, 64)
-		if err != nil {
-			return NumaStat{}, fmt.Errorf("error parsing file %s: %v", entry, err)
-		}
-
-		nodes := make(map[string]int64)
-		for _, nodeEntry := range split[1:] {
-			nodeamount := strings.Split(nodeEntry, "=")
-			if len(nodeamount) != 2 {
-				return NumaStat{}, fmt.Errorf("error parsing file %s", entry)
+	path := filepath.Join(cgroupPath, cgroupEntry)
+	err := sysfs.ParseFileByLines(path,
+		// split line to fields by spaces
+		func(line string) ([]*sysfs.Field, error) {
+			split := strings.Split(line, " ")
+			fields := make([]*sysfs.Field, 0, len(split))
```

Comment:
It's an idiomatic pattern I picked up for slice-construction in a loop: voluntarily pay the price of a single call per iteration to the fast path of the builtin append instead of ending up hard-to find bugs with trailing zero-value entries in case we ever break out early form the loop and (try to) return the items collected so far.

But you're right, with this simple loop it cannot happen. Updated code accordingly.